### ISSUE TITLE
PdfDocument::_parse_string does not handle utf-8 well

### ISSUE THREAD

**Issue by jcppkkk** [cuckoosandbox/cuckoo#1993]
```
Traceback (most recent call last):                                                                
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap                   
    self.run()                                                                                    
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run                          
    self._target(*self._args, **self._kwargs)                                                     
  File "cuckoo/cuckoo/misc.py", line 161, in _worker                        
    conn.send(func(*args, **kwargs))                                                              
  File "cuckoo/cuckoo/processing/static.py", line 1030, in _pdf_worker      
    return PdfDocument(filepath).run()                                                            
  File "cuckoo/cuckoo/processing/static.py", line 725, in run               
    self.walk_object(obj.object, row)                                                             
  File "cuckoo/cuckoo/processing/static.py", line 572, in walk_object       
    entry["urls"].append(self._parse_string(url))                                                 
  File "cuckoo/cuckoo/processing/static.py", line 554, in _parse_string     
    if s.startswith(u"\xfe\xff"):                                                                 
UnicodeDecodeError: 'ascii' codec can't decode byte 0xfe in position 0: ordinal not in range(128) 
```

`startswith` need the pattern and matching string in same coding:
```
>>> u"\xfe\xff".startswith(u"\xfe\xff")
True
>>> "\xfe\xff".startswith("\xfe\xff")
True
>>> "\xfe\xff".startswith(u"\xfe\xff")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xfe in position 0: ordinal not in range(128)
```
maybe try startswith(u"\xfe\xff") first, then fallback to startswith("\xfe\xff") ?

**Comment by jbremer** [comment]
Could you please share a hash? :-)

**Comment by jcppkkk** [comment]
double checked it's f968a86f939e372433c87d6e543683cce4310b29cca74e6553eceae590c0cfbe

**Comment by RicoVZ** [comment]
Hi jcppkkk,

This should also be fixed by the changes I made that I mentioned here: https://github.com/cuckoosandbox/cuckoo/issues/1994. These changes are not yet in the latest Cuckoo release, though.

The changes are in commit: https://github.com/cuckoosandbox/cuckoo/commit/434d8c9afbd0a6ff23e4ff9e0d4687099c685b28

**Comment by jcppkkk** [comment]
It should be fixed by 434d8c9
